tests/test_generate_reports.py

```diff
@@ -0,0 +1,186 @@
+            "/path/to/my/transactions.json",
+        ),
+        (
+            ["--transactions-file-path", "/path/to/my/transactions.json"],
+            "config/user_config.yaml",
+            "/path/to/my/transactions.json",
+        ),
+        (
+            ["-ucfp", "/path/to/my/user_config.yaml"],
+            "/path/to/my/user_config.yaml",
+            "data/merged_transactions.json",
+        ),
+        (
+            ["--user-config-file-path", "/path/to/my/user_config.yaml"],
+            "/path/to/my/user_config.yaml",
+            "data/merged_transactions.json",
+        ),
+    ],
+)
+def test_correct_cli_params(
+    command_params: List[str],
+    user_file_path: str,
+    transactions_file_path: str,
+    cache_user_configuration_mock: Mock,
+    open_mock: Mock,
+    json_mock: Mock,
+) -> None:
+    with patch("personal_finances.generate_reports._write_reports"):
+        runner = CliRunner()
+        result = runner.invoke(generate_reports, command_params)
+        assert result.exit_code == 0
+        cache_user_configuration_mock.assert_called_once_with(user_file_path)
+        open_mock.assert_called_once_with(transactions_file_path, "r")
+
+
+def assert_files_are_equal(file_path: str, result: dict) -> None:
+    with open(file_path, "r") as expected_result:
+        json_result = json.loads(expected_result.read())
+        result_str = json.dumps(result, default=lambda dt_obj: dt_obj.isoformat())
+        assert json_result == json.loads(result_str)
+
+
+def _run_generate_reports(transactions_path: str, configuration_path: str) -> int:
+    runner = CliRunner()
+    return runner.invoke(
+        generate_reports,
+        [
+            "-ucfp",
+            configuration_path,
+            "-tfp",
+            transactions_path,
+        ],
+    ).exit_code
+
+
+def test_generate_reports_output(write_json_mock: Mock) -> None:
+
+    transactions_test_file_path = "tests/test_data/transactions/test_transaction.json"
+    configuration_test_file_path = "tests/test_data/config/test_config_file.yaml"
+
+    exit_code = _run_generate_reports(
+        transactions_test_file_path, configuration_test_file_path
+    )
+    assert exit_code == 0
+
+    result_calls = write_json_mock.call_args_list
+
+    expected_path_name = "reports/1970-01-01T00:00:00+00:00_2100-01-01T00:00:00+00:00/"
+    expected_file_path_prefix = "tests/test_data/expected/"
+    expected_file_name_list = (
+        "balance.json",
+        "income_per_group.json",
+        "income_per_category.json",
+        "income_categorized_transactions.json",
+        "expense_per_group.json",
+        "expense_per_category.json",
+        "expense_categorized_transactions.json",
+    )
+
+    for idx, file_name in enumerate(expected_file_name_list):
+        assert result_calls[idx].args[0] == expected_path_name + file_name
+        assert_files_are_equal(
+            expected_file_path_prefix + file_name, result_calls[idx].args[1]
+        )
```

Comment:
What about:
```
expected_result_path_prefix = (
    "reports/1970-01-01T00:00:00+00:00_2100-01-01T00:00:00+00:00/"
)

expected_content_path_prefix = "tests/test_data/expected/"

expected_file_name_list = (
    "balance.json",
    "income_per_group.json",
    "income_per_category.json",
    "income_categorized_transactions.json",
    "expense_per_group.json",
    "expense_per_category.json",
    "expense_categorized_transactions.json",
)

result_calls = {
    result_call.args[0].removeprefix(expected_result_path_prefix): {
        "file_path": result_call.args[0],
        "file_content": result_call.args[1],
    }
    for result_call in write_json_mock.call_args_list
}

for expected_file_name in expected_file_name_list:
    result_file_path = result_calls[expected_file_name]["file_path"]
    result_file_content = result_calls[expected_file_name]["file_content"]

    assert result_file_path in expected_content_path_prefix + expected_file_name
    assert_file_content_json(
        expected_content_path_prefix + expected_file_name,
        result_file_content
    )
```

It's a bit longer, but it's clearer and it drops the `idx` ordering assumption, which shouldn't matter.